### PR TITLE
added Enable token stream on /completions

### LINKED ISSUE #406: feature: Enable token stream on /completion
**Is your feature request related to a problem? Please describe.**
Currently, `stream: true` is honored only with the `/chat/completion` endpoint

**Describe the solution you'd like**
The `/completion` endpoint should support it as well

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by krishnaduttPanchagnula**
**Description**

This PR fixes #406 

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: api/openai.go [hunk 1]
```diff
@@ -184,6 +184,37 @@ func completionEndpoint(cm *ConfigMerger, o *Option) func(c *fiber.Ctx) error {
 			result = append(result, r...)
 		}
 
+		if input.Stream {
+			responses := make(chan OpenAIResponse)
```
Comment (mudler) [inline api/openai.go:188]: this wouldn't really make it real-time, would it?

see for instance how the `chatCompletion` endpoint tackle this: https://github.com/go-skynet/LocalAI/blob/2c918378652522648bc45dd6b5410c5e1713d9ae/api/openai.go#L261

Comment (krishnaduttPanchagnula) [reply api/openai.go:177]: In my understanding the following code should solve the issue @mudler 

`if input.Stream {
			responses := make(chan OpenAIResponse)

			go func() {
				defer close(responses)
				for _, i := range config.PromptStrings {
					// A model can have a "file.bin.tmpl" file associated with a prompt template prefix
					templatedInput, err := o.loader.TemplatePrefix(templateFile, struct{ Input string }{Input: i})
					if err == nil {
						i = templatedInput
						log.Debug().Msgf("Template found, input modified to: %s", i)
					}

					r, err := ComputeChoices(i, input, config, o.loader, nil, nil)
					if err != nil {
						log.Error().Err(err).Msg("Error computing choices")
						continue
					}

					resp := OpenAIResponse{
						Model:   input.Model,
						Choices: r,
						Object:  "text_completion",
					}
					responses <- resp

				}
			}()

			c.Context().SetBodyStreamWriter(fasthttp.StreamWriter(func(w *bufio.Writer) {
				for ev := range responses {
					var buf bytes.Buffer
					enc := json.NewEncoder(&buf)
					enc.Encode(ev)

					log.Debug().Msgf("Sending chunk: %s", buf.String())
					fmt.Fprintf(w, "data: %v\n", buf.String())
					w.Flush()
				}

				resp := &OpenAIResponse{
					Model:   input.Model, // we have to return what the user sent here, due to OpenAI spec.
					Choices: []Choice{{FinishReason: "stop"}},
				}
				respData, _ := json.Marshal(resp)

				w.WriteString(fmt.Sprintf("data: %s\n\n", respData))
				w.WriteString("data: [DONE]\n\n")
				w.Flush()

			}))

			return nil
		}
`

File: api/openai.go [hunk 2]
```diff
@@ -157,12 +157,77 @@ func completionEndpoint(cm *ConfigMerger, o *Option) func(c *fiber.Ctx) error {
...
+				defer close(responses)
+				for _, i := range config.PromptStrings {
+					// A model can have a "file.bin.tmpl" file associated with a prompt template prefix
+					templatedInput, err := o.loader.TemplatePrefix(templateFile, struct{ Input string }{Input: i})
+					if err == nil {
+						i = templatedInput
+						log.Debug().Msgf("Template found, input modified to: %s", i)
+					}
+
+					r, err := ComputeChoices(i, input, config, o.loader, nil, nil)
```
Comment (mudler) [inline api/openai.go:189]: there is nothing being streamed back here, see: https://github.com/go-skynet/LocalAI/blob/aacb96df7afbd63e1c44f5e0dce580a4714634ec/api/openai.go#L177

**Review by mudler — CHANGES_REQUESTED** [review]
this doesn't look right yet, but almost in the good direction

**Pull request closed without merging**